server/sonar-server/src/main/java/org/sonar/server/permission/ws/DeleteTemplateAction.java

```diff
@@ -75,6 +75,7 @@ private void doHandle(DeleteTemplateWsRequest request) {
       PermissionTemplateDto template = finder.getTemplate(dbSession, newTemplateRef(request.getTemplateId(), request.getTemplateName()));
       checkTemplateUuidIsNotDefault(template.getUuid());
       dbClient.permissionTemplateDao().deleteById(dbSession, template.getId());
+      dbClient.permissionTemplateCharacteristicDao().deleteByTemplateId(dbSession, template.getId());
```

Comment:
I'll update permissionTemplateDao().deleteById() to use the permissionTemplateCharacteristic mapper .